File: user/src/main/java/com/kjj/user/dto/info/UserInfoDto.java
package com.kjj.user.dto.info;

import com.kjj.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class UserInfoDto {
    private Long id;
    private String username;
    private String loginDate;

    public static UserInfoDto from(User user) {
        return new UserInfoDto(
                user.getId(),
                user.getUsername(),
                user.getLoginDate()
        );
    }
}

File: user/src/main/kotlin/com/kjj/user/dto/info/UserInfoDto.kt
package com.kjj.user.dto.info

import com.kjj.user.entity.User
import java.time.LocalDate

data class UserInfoDto(
    val id: Long?,
    val username: String,
    val loginDate: LocalDate?
) {
    companion object {
        fun from(user: User): UserInfoDto {
            return UserInfoDto(
                user.id,
                user.username,
                user.loginDate
            )
        }
    }
}